### ISSUE TITLE
An Action+Route to send the verification email again after expiration

### ISSUE THREAD

**Issue by adazmy** [SymfonyCasts/verify-email-bundle#35]
Just a side note about why there is no action or route inside the expiration error message to ask for a new verification email?

thanks!

**Comment by Tubusy** [comment]
Yes, I've totally failed in trying to create a 'send new verification email' action. Is there an example of how it would be done? I am surprised it is not a part of the bundle as it will _always_ be essential to have one.

**Comment by adazmy** [comment]
If you follow the fact that a user should be logged in to verify his email (debate en #27), then just catch an additionnal ExpiredSignatureException from $this->emailVerifier->handleEmailConfirmation($request, $user); and resend the mail using the code from register.

if you added the email on the URL ( suggestion on #31), get the user object from the email, and resend the mail using the code from register.

**Comment by Tubusy** [comment]
Oh I see, it is as simple as ending a new email. Clever. I was overthinking it. Thanks @adazmy

**Comment by adazmy** [comment]
But still... this should be (in my opinion) implemented directly by the bundle... and maybe have it's own action/route if we want to add this in a message. 
Waiting for the creators to react :)

**Comment by jrushlow** [comment]
Good Morning! Off the top of my head, this is a feature that would be best implemented in the Maker Bundle code. Technically it is possible currently with a little bit of leg work. I'll review the code bases and see how to best implement this over in Maker Bundle.

**Comment by lukepass** [comment]
This feature would be useful too!

**Comment by bastien70** [comment]
Hey! What's new about this idea? I am in the same situation.

When my user tries to log in when he is not activated, I display a notification telling him to activate his account, and I add a "Resend activation email" link.

If I want to be able to manage this, I will be forced, each time a user tries to connect, to generate a new unique link allowing them to access a route that will use the bundle to send a confirmation email.

In my **UserChecker.php** class : 

```php
    public function checkPreAuth(UserInterface $user)
    {

        if (!$user instanceof User) {
            return;
        }

            // $uniqueResendEmailurl = ....
            throw new CustomUserMessageAccountStatusException("Votre compte n'est pas activé. Veuillez confirmer 
            votre inscription en cliquant sur le lien qui vous a été envoyé par email. Pensez à vérifier dans vos spams. <a href=. "$uniqueResendEmailurl" . >Renvoyer l'email de confirmation</a>");
        }
    }
```

The bundle should really be able to integrate this functionality directly.

**Comment by vincentcox** [comment]
@jrushlow , Hi there!
Any idea how to properly implement this? Apart from the controller/twig template that shows this page, and running the function that resends this mail with a new token, there needs to be a check that only allows to send this mail after let's say 2 hours or max 1 mail/hour/user. Otherwise someone can register in your application and fill the mailbox of this user with 100's of verify email links. I saw this happen several times at clients when performing security testing. 

The main issue (a consequence of not using database entries) is that there is no way to keep track of that without making database entries. So I think some code from the "password reset" package needs to be used in this package to make this work. The symfonycasts password reset package also uses a table dedicated to keep track of that and fix the concern above.

Let me know what you think!

**Comment by ToshY** [comment]
I second @vincentcox's idea. The password reset bundle already has most of the logic (and config) which could be used here as well.

**Comment by ToshY** [comment]
@jrushlow It's been quite some time since there was an update on this issue. Could you give as some additional information on what the status is? Is this feature still going to be implemented here / Maker bundle one day? Or should we just implement a workaround ourselves?

**Comment by ToshY** [comment]
**Update**: I've asked this question in the discussions at SymfonyCasts, and Ryan responded that [it will be added to their internal OSS issue tracker](http://disq.us/p/2mr0zph). 🤞

**Comment by ToshY** [comment]
> Good Morning! Off the top of my head, this is a feature that would be best implemented in the Maker Bundle code. Technically it is possible currently with a little bit of leg work. I'll review the code bases and see how to best implement this over in Maker Bundle.

@jrushlow It's been well over a year since you've mentioned in #50 that the functionality was "currently planned". Could you please give us a status update?

**Comment by weaverryan** [comment]
I’d welcome a PR to MakerBundle to generate this and/or any potential helper code in this bundle that would make it easier :)

**Comment by arirangz** [comment]
hello !
any news about this feature?
Thanks

**Comment by Nincha** [comment]
Hey there! 😃 

I'm also interested in an official solution for that, trying to avoid too much non-standard approaches.

Since the mentionned solution with `ExpiredSignatureException` would not work if using the feature added in #27 it is kinda limitating in my case ; has anyone found a way to work around that?

Cheers!
